src/rpc/server.cpp

```diff
@@ -424,6 +432,22 @@ static inline JSONRPCRequest transformNamedArguments(const JSONRPCRequest& in, c
             argsIn.erase(fr);
         } else {
             hole += 1;
+            if (out.params.empty()) initial_hole_size = hole;
+        }
+    }
+    // If leftover "args" param was found, use it as a source of positional
+    // arguments and add named arguments after. This is a convenience for
+    // clients that want to pass a combination of named and positional
+    // arguments, described in doc/JSON-RPC-interface.md#parameter-passing
+    auto positional_args = argsIn.extract("args");
+    if (positional_args && positional_args.mapped()->isArray()) {
+        if (initial_hole_size < std::min<int>(positional_args.mapped()->size(), argNames.size())) {
```

Comment:
The `std::min` confuses me. If `positional_args.mapped()->size() > argNames.size()`, isn't that a different issue than parameters being specified both as positional and named? If we're specifying more positional arguments than the method supports, it doesn't really matter which other named arguments we've provided - it's gonna be an invalid call either way? Additionally, this could be triggered without providing any (non-"arg") named parameters at all, resulting in a misleading error message? If my reasoning is correct, maybe best to split that up into two cases/checks?

Reply:
Your explanation makes sense, thanks. I erroneously understood this line to check for `positional_args` being smaller than `argNames` - whereas actually it checks for `initial_hole_size` being smaller than `argNames`. Since `initial_hole_size` can never be strictly larger than `argNames`, I suppose this is just to check if they're equal in size, which would indicate that not a single named argument was provided (and that `argNames[initial_hole_size]` can't be out of bounds)? In which case, I think it could be helpful to store that in an aptly named boolean just for readability because it's not immediately obvious I think, e.g.

```
bool has_named_arguments{initial_hole_size < (int)argNames.size()};
if (initial_hole_size < (int)positional_args.mapped()->size() && has_named_arguments) {
```

Other than that, I think the error handling of providing too many positional arguments is exactly how it should be as you described, so you can consider this resolved. 👍

Reply:
re: https://github.com/bitcoin/bitcoin/pull/19762#discussion_r1023929056

> I think it could be helpful to store that in an aptly named boolean just for readability

Thanks. Applied your suggestion. I also added unit tests to check for the intended behavior and make sure this doesn't get broken unintentionally in the future.